include/seqan/sequence/string_packed.h

```diff
@@ -579,10 +583,74 @@ resize(
         clear(host(me));
         return 0;
     }
-    TStringSize max_length = (resize(host(me), TTraits::toHostLength(new_length) + 1, tag) - 1) * TTraits::VALUES_PER_HOST_VALUE;
-    if ((TStringSize)new_length > max_length)
-        new_length = max_length;
-    return front(host(me)).i = new_length;
+
+    TStringSize _new_length = new_length;
+    TStringSize max_length = (resize(host(me), TTraits::toHostLength(_new_length) + 1, tag) - 1) * TTraits::VALUES_PER_HOST_VALUE;
+    if (_new_length > max_length)
```

Comment:
When is this true? max_length is the maximum number of values the packed string can hold which should always be greater or equal than the actual length of the packed string, shouldn't it?